test/e2e/performanceprofile/functests/6_mustgather_testing/mustgather.go

```diff
@@ -0,0 +1,174 @@
+package pao_mustgather
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	testutils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	testclient "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/client"
+	"github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/profiles"
+	machineconfigv1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	mustGatherPath = "../../testdata/must-gather/must-gather.e2e"
+)
+
+var mustGatherContentDir string
+var snapshotDir string
+var snapshotName string
+
+var _ = Describe("[rfe_id: 50649] Performance Addon Operator Must Gather", func() {
+	var profile *performancev2.PerformanceProfile
+
+	testutils.BeforeAll(func() {
+		mustgatherPathContent, err := ioutil.ReadDir(mustGatherPath)
+		Expect(err).To(BeNil(), "failed to read the must-gather Directory %s: %v", mustGatherPath, err)
+
+		for _, directory := range mustgatherPathContent {
+			if strings.Contains(directory.Name(), "must-gather") {
+				mustGatherContentDir = filepath.Join(mustGatherPath, directory.Name())
+				break
+			}
+		}
+
+		// pre generated must-gather data
+		// hardcored node name
+		snapshotDir = filepath.Join(mustGatherContentDir, "nodes", "ci-ln-t0prq3t-72292-h4x2n-worker-a-wqnqt")
+		snapshotName = filepath.Join(snapshotDir, "sysinfo.tgz")
+
+		if _, err := os.Stat(snapshotName); errors.Is(err, os.ErrNotExist) {
+			Expect(err).ToNot(HaveOccurred(), "failed to read sysinfo.tgz file %s: %v", snapshotName, err)
+		}
+
+		err = Untar(snapshotDir, snapshotName)
```

Comment:
how do we do cleanup?

Reply:
done

Reply:
how was this addressed?

Reply:
here, https://github.com/openshift/cluster-node-tuning-operator/blob/7e46efaa55dcd2ba972839a4cf27a47b7694026b/test/e2e/performanceprofile/functests/6_mustgather_testing/test_suite_mustgather_test.go#L40

maybe it could be better but I think is enough for this PR

Reply:
I think it's fine to do this in the aftersuite. It would be better to untar into a randomly generated temporary folder (https://pkg.go.dev/os#TempDir) and remove the temporary directory straight away when the suite ends.
I think this is a further step we should do for safety and robustness (and to minimize the risk we clog the disk of the CI boxes).
In the future we can move all the snapshot handling at the same level - make temp dir and untar in beforesuite, clean up (just remove the temp dir and all its content) on aftersuite